SynthSeg/training_with_tfrecords.py

```diff
@@ -206,16 +317,22 @@ def build_callbacks(
     wandb: bool = False,
     wandb_log_freq: Union[str, int] = "epoch",
     training_opts: Optional[TrainingOptions] = None,
+    num_gpu: int = None, 
+    steps_per_epoch: int = None
 ):
     # create log folder
     log_dir = output_dir / "logs"
     log_dir.mkdir(exist_ok=True)
 
     # model saving callback
-    save_file_name = os.path.join(output_dir, "%s_{epoch:03d}.h5" % metric_type)
-    # save_weights_only=True is a workaround when training with mixed precision: https://github.com/keras-team/tf-keras/issues/203
-    # For now, setting this to true disables the posibility to train a model further, at least if we dont want to reinit the momentum ...
-    callbacks = [tf.keras.callbacks.ModelCheckpoint(save_file_name, verbose=1, save_weights_only=True)]
+    save_file_name = os.path.join(output_dir, "%s_{epoch:03d}.keras" % metric_type)
+
+
+    tracking_file = os.path.join(output_dir, f"time_recorder-num_gpu_{num_gpu}-strategy-{training_opts.strategy}_scaling-{training_opts.scaling_type}_steps_per_epoch-{steps_per_epoch}.csv")
+    tracking_callback = TimingCallback(tracking_file)
```

Comment:
Again, i would prefer to leave this callback out of this PR.

Reply:
done